internal/lib/container_server.go

```diff
@@ -333,16 +333,36 @@ func (c *ContainerServer) LoadSandbox(id string) error {
 	sb.SetSeccompProfilePath(spp)
 	sb.SetNamespaceOptions(&nsOpts)
 
-	// We add a netNS only if we can load a permanent one.
+	// We add an NS only if we can load a permanent one.
 	// Otherwise, the sandbox will live in the host namespace.
-	if c.config.ManageNetworkNSLifecycle {
-		netNsPath, err := configNetNsPath(&m)
+	if c.config.ManageNSLifecycle {
+		netNsPath, err := configNsPath(&m, rspec.NetworkNamespace)
 		if err == nil {
-			nsErr := sb.NetNsJoin(netNsPath, sb.Name())
+			nsErr := sb.NetNsJoin(netNsPath)
 			// If we can't load the networking namespace
 			// because it's closed, we just set the sb netns
 			// pointer to nil. Otherwise we return an error.
-			if nsErr != nil && nsErr != sandbox.ErrClosedNetNS {
+			if nsErr != nil && nsErr != sandbox.ErrClosedNS {
+				return nsErr
+			}
+		}
+		ipcNsPath, err := configNsPath(&m, rspec.IPCNamespace)
+		if err == nil {
+			nsErr := sb.IpcNsJoin(ipcNsPath)
+			// If we can't load the IPC namespace
+			// because it's closed, we just set the sb ipcns
+			// pointer to nil. Otherwise we return an error.
+			if nsErr != nil && nsErr != sandbox.ErrClosedNS {
+				return nsErr
+			}
```

Comment:
@haircommander is there any use-case behind the check for closed namespaces? I'm not sure if we really need this. If the namespace is closed I assume that something went wrong and mark the sandbox as not-ready that it gets properly re-created.

Reply:
hm good question. to be honest I just copied this from the old net ns managing code. There are fewer calls to close now though.

Reply:
This might be the root cause that we tear down the local loopback interface when the namespace is not correctly set-up (referring to https://github.com/openshift/containernetworking-plugins/pull/15 and https://bugzilla.redhat.com/show_bug.cgi?id=1754154#c11).

Ref: https://github.com/cri-o/cri-o/pull/3086